server/portal/apps/projects/managers/base.py

```diff
@@ -181,6 +181,32 @@ def list(self, offset=0, limit=100):
             limit=limit
         )]
 
+    def apply_permissions(self, project, username, acl=None):
+        """Index project and update acls
+        """
+        project_id = project.project_id
+        project_indexer.apply_async(args=[project_id])
+        project_root = project.storage.storage.root_dir
+        if acl == 'add':
+            self._add_acls(username, project_id, project_root)
+        elif acl == 'remove':
+            self._remove_acls(username, project_id, project_root)
+
+    def transfer_ownership(self, project_id, old_owner, new_owner):
+        """Transfer ownership by setting new PI
+        and demoting old PI to Co-PI
+        """
+        old_pi = get_user_model().objects.get(username=old_owner)
+        new_pi = get_user_model().objects.get(username=new_owner)
+        prj = Project(
+            service_account(),
+            project_id
+        )
+        prj.transfer_pi(old_pi, new_pi)
+        self.apply_permissions(prj, new_owner)
+        self.apply_permissions(prj, old_owner)
```

Comment:
If `username` is not referenced in `apply_permissions` when no acl is provided, can this be deleted?
```suggestion
```

Reply:
No, we still need indexing to take place.

Reply:
If so, I'd suggest we call the indexer directly here